File: app/src/main/java/eu/napcode/gonoteit/repository/notes/results/NotesResult.java
package eu.napcode.gonoteit.repository.notes.results;

import android.arch.lifecycle.LiveData;
import android.arch.paging.PagedList;

import eu.napcode.gonoteit.model.note.NoteModel;
import eu.napcode.gonoteit.repository.Resource;

public class NotesResult {

    LiveData<PagedList<NoteModel>> notes;
    LiveData<Resource> resource;

    public NotesResult(LiveData<PagedList<NoteModel>> notes, LiveData<Resource> resource) {
        this.notes = notes;
        this.resource = resource;
    }

    public LiveData<PagedList<NoteModel>> getNotes() {
        return notes;
    }

    public LiveData<Resource> getResource() {
        return resource;
    }
}

File: app/src/main/java/eu/napcode/gonoteit/data/notes/results/NotesResult.kt
package eu.napcode.gonoteit.data.notes.results

import android.arch.lifecycle.LiveData
import android.arch.paging.PagedList

import eu.napcode.gonoteit.model.note.NoteModel
import eu.napcode.gonoteit.repository.Resource

class NotesResult(notes: LiveData<PagedList<NoteModel>>, resource: LiveData<Resource<*>>) {

    var notes: LiveData<PagedList<NoteModel>>
        internal set

    var resource: LiveData<Resource<*>>
        internal set

    init {
        this.notes = notes
        this.resource = resource
    }
}
